charms/worker/k8s/src/charm.py

```diff
@@ -124,8 +122,8 @@ def _distribute_cluster_tokens(self, relation, _role):
             if app_databag.get(unit.name):
                 continue
 
-            name = unit.name.replace("/", "-")
-            token = self.api_manager.create_join_token(name)
+            name = platform.node()
+            token = self.api_manager.create_join_token(name, worker=self.is_worker)
```

Comment:
Can I use the same token generated by a control-plane node to add either a worker or a control-plane?  If not, distribute tokens needs some work

I expected this method `_distribute_cluster_tokens`  to be called twice by the leader control-plane unit
1) with the peer relation from `self.model.get_relation("cluster")` to generate control-plane tokens
2) with the worker relation from the `self.model.get_relation("k8s-cluster")`  to generate worker tokens

you want to share control-plane tokens over the `cluster` peer relation
you want to share worker tokens over the `k8s-cluster` app relation